src/api/db/migrate/20170426153510_add_references_to_review.rb

```diff
@@ -0,0 +1,32 @@
+class AddReferencesToReview < ActiveRecord::Migration[5.0]
+  def up
+    start = Time.now
+    add_reference(:reviews, :entity, polymorphic: true)
+
+    tmp = Review.where("by_user is not null").distinct(:by_user).pluck(:by_user)
+    User.where(login: tmp).each do |user|
+      execute "UPDATE reviews SET entity_id = #{user.id}, entity_type = 'User' WHERE by_user = '#{user.login}'"
+    end
```

Comment:
This is slower than the iteration as we reduce the number of users with the distinct to only 500 and don't need to join 100.000 records.